tracer.go

```diff
@@ -235,7 +235,7 @@ func (t *Tracer) startSpanWithOptions(
 		ctx.spanID = SpanID(ctx.traceID.Low)
 		ctx.parentID = 0
 		ctx.flags = byte(0)
-		if hasParent && parent.isDebugIDContainerOnly() {
+		if hasParent && parent.isDebugIDContainerOnly() && t.isDebugAllowed(operationName) {
```

Comment:
@isaachier I think I forgot this case, if the debug trace is started via using headers, the tag will be "jaeger-debug-id". I guess we could just add the "sampling.priority" tag here too...

Reply:
You mean an extracted span contains that header or a client locally used headers to start debug tracing somehow. I don't really understand the latter.